kamelet-support/src/test/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletBindingStepParserServiceTest.java

```diff
@@ -128,4 +128,58 @@ void appliesToInvalid(String resourcePath) throws IOException {
                 .readAllBytes(), StandardCharsets.UTF_8);
         assertThat(dslSpecification.appliesTo(input)).isFalse();
     }
+
+    @Test
+    void parseNullSource() throws IOException {
+        String input = new String(
+                Objects.requireNonNull(
+                        this.getClass().getResourceAsStream("null-source.binding.yaml"))
+                .readAllBytes(), StandardCharsets.UTF_8);
+        StepParserService.ParseResult<Step> parsed = dslSpecification.getStepParserService().deepParse(input);
+        assertThat(parsed.getSteps()).hasSize(1);
+        assertThat(parsed.getSteps().get(0))
+                .extracting(Step::getName)
+                .isEqualTo("log-sink");
+        assertThat(parsed.getSteps().get(0))
+                .extracting(Step::getType)
+                .isEqualTo("END");
+        String parsedYaml = dslSpecification.getDeploymentGeneratorService()
+                .parse(parsed.getSteps(), parsed.getMetadata(),
+                        parsed.getParameters());
+        assertThat(parsedYaml).isEqualToNormalizingNewlines(input);
+    }
+
+    @Test
+    void parseNullSink() throws IOException {
+        String input = new String(
+                Objects.requireNonNull(
+                        this.getClass().getResourceAsStream("null-sink.binding.yaml"))
+                .readAllBytes(), StandardCharsets.UTF_8);
+        StepParserService.ParseResult<Step> parsed = dslSpecification.getStepParserService().deepParse(input);
+        assertThat(parsed.getSteps()).hasSize(1);
+        assertThat(parsed.getSteps().get(0))
+                .extracting(Step::getName)
+                .isEqualTo("timer-source");
+        assertThat(parsed.getSteps().get(0))
+                .extracting(Step::getType)
+                .isEqualTo("START");
+        String parsedYaml = dslSpecification.getDeploymentGeneratorService()
+                .parse(parsed.getSteps(), parsed.getMetadata(),
+                        parsed.getParameters());
+        assertThat(parsedYaml).isEqualToNormalizingNewlines(input);
+    }
+
+    @Test
+    void parseNullSourceNullSink() throws IOException {
+        String input = new String(
+                Objects.requireNonNull(
+                        this.getClass().getResourceAsStream("null-source-null-sink.binding.yaml"))
+                .readAllBytes(), StandardCharsets.UTF_8);
+        StepParserService.ParseResult<Step> parsed = dslSpecification.getStepParserService().deepParse(input);
+        assertThat(parsed.getSteps()).hasSize(0);
```

Comment:
Is this a big deal? :thinking: 

https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=746&id=KaotoIO_kaoto-backend&open=AYkdVe_-Ber657K2ULFM

Reply:
It's supposed to be more optimized, I guess.

Reply:
I guess it is worth adding that to the PR.

Reply:
done :+1: